analyses/stackanalyses/sa.go

```diff
@@ -0,0 +1,186 @@
+	if err != nil {
+		log.Fatal().Err(err).Msgf(err.Error())
+	}
+	_, err = io.Copy(fw, fd)
+	if err != nil {
+		log.Fatal().Err(err).Msgf(err.Error())
+	}
+	_ = writer.WriteField("ecosystem", fileStats.Eco)
+	_ = writer.WriteField("file_path", "/tmp/bin")
+	err = writer.Close()
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Error closing Buffer Writer in SA Request.")
+	}
+	apiResponse := utils.HTTPRequestMultipart(requestData, writer, manifest)
+	body := validatePostResponse(apiResponse)
+	log.Debug().Msgf("Success: saPostRequest.")
+	return body
+}
+
+// saGetRequest performs Stack Analyses GET Request to CRDA Server.
+func saGetRequest(requestParams SARequestType, saPost SAPostResponseType, back *backoff.Backoff) SAGetResponseType {
+	log.Debug().Msgf("Executing: saGetRequest.")
+	requestData := utils.HTTPRequestType{
+		Method:          http.MethodGet,
+		Endpoint:        utils.APISA + "/" + saPost.ID,
+		ThreeScaleToken: requestParams.ThreeScaleToken,
+		Host:            requestParams.Host,
+	}
+	d := back.Duration()
+	log.Debug().Msgf("Sleeping for %s", d)
+	time.Sleep(d)
+	apiResponse := utils.HTTPRequest(requestData)
+	if apiResponse.StatusCode == http.StatusAccepted {
+		// Retry till server respond 200 or Timeout Error or Exponential Backoff limit hit.
+		log.Debug().Msgf("Retying...")
+		saGetRequest(requestParams, saPost, back)
+	}
+	body := validateGetResponse(apiResponse)
+	return body
+}
+
+// validateResponse validates API Response.
+func validatePostResponse(apiResponse *http.Response) SAPostResponseType {
+	log.Debug().Msgf("Executing validatePostResponse.")
+	var body SAPostResponseType
+	err := json.NewDecoder(apiResponse.Body).Decode(&body)
+	if apiResponse.StatusCode != http.StatusOK {
+		log.Debug().Msgf("Status from Server: %d", apiResponse.StatusCode)
+		log.Fatal().Err(err).Msgf(err.Error())
+	}
+	log.Debug().Msgf("Success validatePostResponse.")
+	return body
+}
+
+// validateGetResponse validates API Response.
+func validateGetResponse(apiResponse *http.Response) SAGetResponseType {
+	log.Debug().Msgf("Executing validateGetResponse.")
+	var body SAGetResponseType
+	err := json.NewDecoder(apiResponse.Body).Decode(&body)
+	if apiResponse.StatusCode != http.StatusOK {
+		log.Debug().Msgf("Status from Server: %d", apiResponse.StatusCode)
+		log.Fatal().Err(err).Msgf(err.Error())
+	}
+	log.Debug().Msgf("Success validateGetResponse.")
+	return body
+}
+
+// readManifest Manifest File validator and reader.
+func readManifest(shellPath string, manifestFile string) ReadManifestResponse {
+	log.Debug().Msgf("Executing readManifest.")
+	stats, err := os.Stat(manifestFile)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Error")
+	}
+	fileStats := ReadManifestResponse{
+		RawFileName: stats.Name(),
+	}
+	msg := fmt.Sprintf("Support for %s is coming soon. Thanks for your Patience. :)", fileStats.RawFileName)
+	switch fileStats.RawFileName {
+	case "requirements.txt":
```

Comment:
python manifest file shall be in any of the following pattern,

```
               "filenamePatterns": [
                    "*-requirements.txt",
                    "requirements-*.txt",
                    "constraints-*.txt",
                    "*-constraints.txt",
                    "*-requirements.in",
                    "requirements-*.in"
```
Reference: https://github.com/microsoft/vscode-python/blob/91ff97f758e82387f9761a3b8ab78c71e55b76be/package.json#L1861

Though our extension limited to requirements.txt, the new implementation should be limited to one file pattern.

Reply:
> Though our extension limited to requirements.txt, the new implementation should be limited to one file pattern.

That's what. We are limiting to `requirements.txt` only. I could not understand the ask here.

Reply:
Typo, I meant shouldn't be limited to. :)

Reply:
Okay, I am not sure if Flow for other formats for generating `pylist.json` out would be same as `requirements.txt` ..? 

Do we support `requirements.in` like file Formats in our extension so..?

Reply:
yes, it will have same format.

Reply:
Ack.